src/averageAges.js

```diff
@@ -55,10 +80,33 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // 1. find a mother of each person (or only for men)
-  // 2. keep people who have mothers in the array
-  // 3. calculate the difference child.born - mother.born
-  // 4. return the average value
+  const mothersWithChildren = people.reduce((arr, person) => {
+    const childrenBorn = people.filter((child) => {
+      const hasMother = person.name === child.mother;
+
+      return onlyWithSon ? child.sex === 'm' && hasMother : hasMother;
+    }).map((child) => child.born);
+
+    if (!childrenBorn.length) {
```

Comment:
The 'if' statement is unnecessary. If 'childrenBorn.length' is zero, the spread operator will add nothing to 'arr', so you can always return '[...arr, { motherBorn: person.born, childrenBorn }]' without checking the length first.